feat: add git worktree support to create-branch command

TellMeAlex opened the pull request:
- Add --worktree and --worktree-path flags to create-branch command
- Implement CreateWorktree, ListWorktrees, and RemoveWorktree in GitProvider
- Add Worktree domain model to represent git worktree state
- Add createWorktreeBranch use case method with default path generation
- Add comprehensive unit tests for git worktree operations
- Add integration tests for worktree creation workflow
- Update documentation with worktree usage examples and benefits
- Support fork context (upstream/origin) for worktree creation

Closes #146

TellMeAlex committed:
feat: add git worktree support to create-branch command

- Add --worktree and --worktree-path flags to create-branch command
- Implement CreateWorktree, ListWorktrees, and RemoveWorktree in GitProvider
- Add Worktree domain model to represent git worktree state
- Add createWorktreeBranch use case method with default path generation
- Add comprehensive unit tests for git worktree operations
- Add integration tests for worktree creation workflow
- Update documentation with worktree usage examples and benefits
- Support fork context (upstream/origin) for worktree creation

Closes #146

danielfn commented:
@TellMeAlex thank you for your contribution!

Not sure why, but the PR-verify workflow did not run with your commit. Trying it locally, I see there are errors with gofmt, could you fix them (`make fmt`) and push again to trigger the CI?

TellMeAlex committed:
fix: apply gofmt formatting

TellMeAlex committed:
chore: revert tool-versions to original golang plugin

TellMeAlex commented:
Hi @danielfn, thanks for the feedback\! I've already applied `gofmt` formatting and pushed the changes. However, I notice the CI is still failing. Could you help me identify what else might need to be fixed?

danielfn commented:
> Hi @danielfn, thanks for the feedback! I've already applied `gofmt` formatting and pushed the changes. However, I notice the CI is still failing. Could you help me identify what else might need to be fixed?

@TellMeAlex there is a failure in the tests, mocks may need to be updated:
```
-: # github.com/InditexTech/gh-sherpa/internal/fork [github.com/InditexTech/gh-sherpa/internal/fork.test]
Error: internal/fork/fork_test.go:116:43: cannot use gitProvider (variable of type *mockGitProvider) as domain.GitProvider value in argument to NewManager: *mockGitProvider does not implement domain.GitProvider (missing method CreateWorktree)
```

TellMeAlex committed:
fix: add missing CreateWorktree, ListWorktrees, and RemoveWorktree methods to GitProvider mocks

- Added the three new methods to both manual mock in fork_test.go and autogenerated mock_GitProvider.go
- Fixes CI failure in PR #147 caused by interface extension in commit 2b06ee7

TellMeAlex committed:
Merge branch 'main' into feature/GH-146-add-support-for-creating-gi

TellMeAlex committed:
feat: add git worktree support to create-branch command

- Add --worktree and --worktree-path flags to create-branch command
- Implement CreateWorktree, ListWorktrees, and RemoveWorktree in GitProvider
- Add Worktree domain model to represent git worktree state
- Add createWorktreeBranch use case method returning the created worktree path
- Add comprehensive unit tests for git worktree operations
- Support fork context (upstream/origin) for worktree creation
- Fix test isolation bug in CreateBranchWorktreeTestSuite (SetupSubTest vs SetupTest)
- Fix default worktree path assertion to match the fake repository name

Closes #146

Signed-off-by: Alejandro de la Fuente de la Rosa <alejandrodlfdlr@ext.inditex.com>
